### PR TITLE
Allow custom options for SearchInput JS

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

I'm working on something that may use the existing JS code used by the new filtering/tagging system used by the Kanban and it would require inheriting several options from the original input.
I added support for adding custom attributes, data attributes, and classes either statically or by copying them directly from the original input element.

JS tests were updates and Kanban filter was manually tested again just be be sure I didn't somehow break it.

### CODE REVIEW

File: js/modules/SearchTokenizer/SearchInput.js [hunk 1]
```diff
@@ -55,20 +59,64 @@ export default class SearchInput {
...
+        }, new_attrs));
+
+        // Apply attributes including data attributes
+        this.displayed_input.attr(new_attrs);
+
+        // Apply classes
+        if (Array.isArray(this.options.input_options.classes)) {
+            this.displayed_input.addClass(this.options.input_options.classes.join(' '));
+        } else if (this.options.input_options.classes === 'copy') {
+            this.displayed_input.addClass(this.original_input.attr('class').join(' '));
```
Comment (cedric-anne) [inline js/modules/SearchTokenizer/SearchInput.js:116]: `$.attr('class')` will return a string.
```suggestion
            this.displayed_input.addClass(this.original_input.attr('class'));
```

File: tests/js/SearchTokenizer/SearchInput.test.js [hunk 2]
```diff
@@ -58,6 +58,73 @@ describe('Search Tokenizer Input', () => {
...
+        $.each(input_options.attributes, (key, value) => {
+            test_input.attr(key, value);
+        });
+        $.each(input_options.data, (key, value) => {
+            test_input.attr('data-' + key, value);
+        });
+        test_input.addClass(input_options.classes);
+
+        const search_input = new SearchInput(test_input, {
+            input_options: input_options
```
Comment (cedric-anne) [inline tests/js/SearchTokenizer/SearchInput.test.js:114]:
```suggestion
            input_options: {
                classes: 'copy',
                attributes: 'copy',
                data: 'copy',
            }
```

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]
Not related to this pr (and this should be done separatly), but  typing in user or group filter with spaces characters like "Exploit N3" is very hard. I can use quotes but at edition of the filter, they are removed

**Comment by cconard96** [comment]
> Not related to this pr (and this should be done separatly), but typing in user or group filter with spaces characters like "Exploit N3" is very hard. I can use quotes but at edition of the filter, they are removed

Noted.

**Pull request merged**